### DISCUSSION TITLE
repository如何分表更新？

### DISCUSSION THREAD

**Discussion by tiansfather** [dotnetcore/FreeSql#815] [General]
repository分表后，使用toupdate仍然更新的是主表
`
var sql1=manager.Repository.Where(o => o.UserId == CurrentSession.Id && o.Group == group).ToSql();
var sql2=manager.Repository.Where(o => o.UserId == CurrentSession.Id && o.Group == group).ToUpdate().Set(o => o.Readed, true).ToSql();
`

sql1
"SELECT a.`Id`, a.`CreatorUserId`, a.`CreatorName`, a.`CreationTime`, a.`TenantId`, a.`UserId`, a.`Group`, a.`Type`, a.`Title`, a.`Content`, a.`Url`, a.`Pic`, a.`Readed`, a.`Items` \r\nFROM `Message_1` a \r\nWHERE (a.`UserId` = 1 AND a.`Group` = '任务') AND (a.`TenantId` = 1)"
sql2
"UPDATE `Message` SET `Readed` = ?p_0 \r\nWHERE (`Id` in (select * from (SELECT a.`Id` \r\nFROM `Message_1` a \r\nWHERE (a.`UserId` = 1 AND a.`Group` = '任务') AND (a.`TenantId` = 1)) ftb_upd)) AND (`TenantId` = 1)"

**Comment by luoyunchong** [comment]
http://freesql.net/guide/sharding.html

**Comment by 2881099** [comment]
.ToUpdate()
.AsTable(...) //先这样处理，下个版本解决
.Set(o => o.Readed, true).ToSql();
